### PR TITLE
fix #3978: importing all duplicates in Windows

### LINKED ISSUE #3978: Re-importing a folder ignores second xmp files created for duplicates
Using the Duplicate function in lighttable mode creates a second xmp file with _01 in the file name. If the photo or its folder is removed under Collect Images, then re-imported, darktable only recognizes the original xmp file and not the second xmp made from the duplicate.  

Darktable Version: 3.0
Windows 10
OpenCL  not activated

**Based on comments below and on pixls.us this seems to be a problem just on the Windows version**

### PR DESCRIPTION
**Pull request by Mark-64**
Fix for #3978 
Also fixed a related bug, leaving .xmp files behind when trashing images

### CODE REVIEW

File: src/common/image.c [hunk 1]
```diff
@@ -999,8 +1005,39 @@ void dt_image_read_duplicates(const uint32_t id, const char *filename)
...
+			
+					while ((c3 <= c4) && valid_filename)
+					{
+						if (!( *c3 >= '0' && *c3 <= '9' )) valid_filename = false;
+						c3++;
+					}
+				
+				}
+				if (valid_filename) files = g_list_append(files, g_build_filename(imgpath, file, NULL));
+				g_free(file);
```
Comment (TurboGit) [inline src/common/image.c]: This code is duplicated below. I would like to see that put into a separate routine maybe src/win/filepath.c ?

Also not that we do not use bool (C++) but gboolean and there is no space after if, while.

File: src/win/filepath.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2009--2012 johannes hanika.
+    copyright (c) 2010--2012 tobias ellinghaus.
```
Comment (TurboGit) [inline src/win/filepath.c]: Please remove those lines. You can probably add your name here but I feel it should better to remove all copyright in the sources. Does not makes sense at this time almost no one update them.

File: src/win/filepath.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2009--2012 johannes hanika.
+    copyright (c) 2010--2012 tobias ellinghaus.
```
Comment (TurboGit) [inline src/win/filepath.h]: Likewise.

File: src/common/image.c [hunk 4]
```diff
@@ -999,8 +1007,8 @@ void dt_image_read_duplicates(const uint32_t id, const char *filename)
       do
       {
         char *file = g_utf16_to_utf8(data.cFileName, -1, NULL, NULL, NULL);
-        files = g_list_append(files, g_build_filename(imgpath, file, NULL));
-        g_free(file);
+				if(win_valid_duplicate_filename(file)) files = g_list_append(files, g_build_filename(imgpath, file, NULL));
```
Comment (TurboGit) [inline src/common/image.c]: Looks like too much spaces here or tabulations are used?

**Comment by Mark-64** [comment]
changes requested implemented

**Comment by TurboGit** [comment]
Thanks  for the first contribution ! And you are fixing a long standing bug, congrats!

**Comment by TurboGit** [comment]
Also merged in 3.0.1 branch.

**Pull request merged**